src/Symfony/Component/Workflow/Tests/DependencyInjection/WorkflowGuardListenerPassTest.php

```diff
@@ -0,0 +1,103 @@
+        $this->container->register('security.authentication.trust_resolver', AuthenticationTrustResolverInterface::class);
+        $this->container->register('security.role_hierarchy', RoleHierarchy::class);
+        $this->container->register('validator', ValidatorInterface::class);
+
+        $this->compilerPass->process($this->container);
+
+        $this->assertFalse($this->container->hasParameter('workflow.has_guard_listeners'));
+    }
+
+    public function testExceptionIfTheTokenStorageServiceIsNotPresent()
+    {
+        $this->expectException(LogicException::class);
+        $this->expectExceptionMessage('The "security.token_storage" service is needed to be able to use the workflow guard listener.');
+        $this->container->setParameter('workflow.has_guard_listeners', true);
+        $this->container->register('security.authorization_checker', AuthorizationCheckerInterface::class);
+        $this->container->register('security.authentication.trust_resolver', AuthenticationTrustResolverInterface::class);
+        $this->container->register('security.role_hierarchy', RoleHierarchy::class);
+
+        $this->compilerPass->process($this->container);
+    }
+
+    public function testExceptionIfTheAuthorizationCheckerServiceIsNotPresent()
+    {
+        $this->expectException(LogicException::class);
+        $this->expectExceptionMessage('The "security.authorization_checker" service is needed to be able to use the workflow guard listener.');
+        $this->container->setParameter('workflow.has_guard_listeners', true);
+        $this->container->register('security.token_storage', TokenStorageInterface::class);
+        $this->container->register('security.authentication.trust_resolver', AuthenticationTrustResolverInterface::class);
+        $this->container->register('security.role_hierarchy', RoleHierarchy::class);
+
+        $this->compilerPass->process($this->container);
+    }
+
+    public function testExceptionIfTheAuthenticationTrustResolverServiceIsNotPresent()
+    {
+        $this->expectException(LogicException::class);
+        $this->expectExceptionMessage('The "security.authentication.trust_resolver" service is needed to be able to use the workflow guard listener.');
+        $this->container->setParameter('workflow.has_guard_listeners', true);
+        $this->container->register('security.token_storage', TokenStorageInterface::class);
+        $this->container->register('security.authorization_checker', AuthorizationCheckerInterface::class);
+        $this->container->register('security.role_hierarchy', RoleHierarchy::class);
+
+        $this->compilerPass->process($this->container);
+    }
+
+    public function testExceptionIfTheRoleHierarchyServiceIsNotPresent()
+    {
+        $this->expectException(LogicException::class);
+        $this->expectExceptionMessage('The "security.role_hierarchy" service is needed to be able to use the workflow guard listener.');
```

Comment:
I suggest moving those 2 `expect` calls just before the `process` call, to ensure that the exception is expected in that call and not in the setup logic before it (same in other methods)